File: learn the basics/hashing/hashmap.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

int main() {
    // Creating a hash map (unordered_map in C++)
    unordered_map<string, int> hash_map;

    // Adding key-value pairs
    hash_map["apple"] = 5;
    hash_map["banana"] = 10;

    // Accessing values by key
    cout << hash_map["apple"] << endl;  // Output: 5

    // Checking if a key exists
    if (hash_map.find("banana") != hash_map.end()) {
        cout << "Banana is in the hash map." << endl;
    }

    // Removing a key-value pair
    hash_map.erase("apple");

    // Iterating over key-value pairs
    for (const auto& pair : hash_map) {
        cout << pair.first << ": " << pair.second << endl;
    }

    return 0;
}

File: learn the basics/hashing/hashmap.py
array = [10, 20, 10, 30, 20, 30, 40, 10, 50, 30, 20, 10, 40, 50, 50, 20, 30]
hash_map = {}
for i in array:
    hash_map[i] = hash_map.get(i,0)+1 #here get first insitialize the value of the key as 0 then increment it with +1
print(hash_map)